BEIMA.Backend/MongoService/BaseDevice.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace BEIMA.Backend.MongoService
+{
+    /// <summary>
+    /// This class represents a Device. This object contains all the required fields necessary
+    /// for a device. This is meant to be used to convert data received from an endpoint, back into a BSON object.
+    /// </summary>
+    public class BaseDevice
+    {
+
+        /// <summary>
+        /// Empty constructor, this allows for a device to be instantiated through Object Initializers.
+        /// </summary>
+        public BaseDevice()
+        {
+            Fields = new BsonDocument();
+            Location = new BsonDocument();
+            LastModified = new BsonDocument();
+        }
+
+        /// <summary>
+        /// Full constructor to instantiate a Device object.
+        /// </summary>
+        /// <param name="id">ObjectId of the object itself.</param>
+        /// <param name="deviceTypeId">ObjectId of the DeviceType this Device is linked to.</param>
+        /// <param name="deviceTag"></param>
+        /// <param name="manufacturer"></param>
+        /// <param name="modelNum"></param>
+        /// <param name="serialNum"></param>
+        /// <param name="yearManufactured"></param>
+        /// <param name="notes"></param>
+        public BaseDevice(ObjectId id, ObjectId deviceTypeId, string deviceTag, string manufacturer, string modelNum, string serialNum, int yearManufactured, string notes)
+        {
+            //BsonClassMap.RegisterClassMap<BaseDevice>();
+
+            Id = id;
+            DeviceTypeId = deviceTypeId;
+            DeviceTag = deviceTag ?? string.Empty;
+            Manufacturer = manufacturer ?? string.Empty;
+            ModelNum = modelNum ?? string.Empty;
+            SerialNum = serialNum ?? string.Empty;
+            YearManufactured = yearManufactured;
```

Comment:
Should we make this a nullable int? Doesn't seem like we should be requiring every device to have this instantiated.

Reply:
I'm not sure if MongoDB allows nullable ints, I will look into this.

Reply:
If anything, we could have it be set to -1 or something.

Reply:
MongoDB can store null values, but there is no way to truly specify it as an int if it is null. I am just going to store -1 as the default value as an alternative.

Reply:
Resolved [401e45f](https://github.com/cs481-ekh/s22-beima/pull/114/commits/401e45f2842b9ab766d956b9ff5b098c2098b2a0)